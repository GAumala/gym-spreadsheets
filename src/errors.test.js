const { SheetBoundaryError } = require('./errors.js');
const { memberArraySchema } = require('./schemas.js');
const { translateMemberKey } = require('./boundary.js');

const createBoundaryError = (docInfo, schema, data) => {
  const validation = schema.validate(data);
  return new SheetBoundaryError(docInfo, validation.error);
}

describe('SheetBoundaryError', () => {
  it('uses Joi error data to create empty string error msg', () => {
    const docInfo = {
      docTitle: 'Test Document',
      sheetTitle: '',
      rowCount: 2,
      translateKey: translateMemberKey
    };

    const data = [
      {id: '', 
       nombre: 'Carlos Hernández', 
       email: '', 
       lesiones: '', 
       entrada: '07:00'},
      {id: '', 
       nombre: '', 
       email: '', 
       lesiones: '', 
       entrada: '07:00'}
    ];
    const error = createBoundaryError(docInfo, memberArraySchema, data);
    expect(error.message).toMatchSnapshot();
  });

it('uses Joi error data to create excluded value error msg', () => {
    const docInfo = {
      docTitle: 'Test Document',
      sheetTitle: '',
      rowCount: 2,
      translateKey: translateMemberKey
    };

    const data = [
      {id: '', 
       nombre: 'Carlos Hernández', 
       email: '', 
       lesiones: '', 
       entrada: '13:00'},
      {id: '', 
       nombre: 'Víctor Borja', 
       email: '', 
       lesiones: '', 
       entrada: '07:00'}
    ];
    const error = createBoundaryError(docInfo, memberArraySchema, data);
    expect(error.message).toMatchSnapshot();
  });
});
