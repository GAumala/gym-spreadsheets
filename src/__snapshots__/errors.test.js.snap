// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SheetBoundaryError uses Joi error data to create empty string error msg 1`] = `
"Por favor revisa los datos ingresados en el documento 'Test Document'.
La columna 'NOMBRE' de la fila #3 no debe estar vac√≠a."
`;

exports[`SheetBoundaryError uses Joi error data to create excluded value error msg 1`] = `
"Por favor revisa los datos ingresados en el documento 'Test Document'.
La columna 'ENTRADA' solo permite valores [06:00,07:00,08:00,09:30,11:00,12:00,17:00,18:00,19:00], pero la fila #2 contiene '13:00'."
`;
